### PR TITLE
potential solution for cyclic dependency errors in complex card relationships

### PR DESCRIPTION
**Pull request by burieberry**
Came across this error in demo-cards, when trying to add `Booking` card as a field in `Pet` card:
error
The issue was, `Booking` card has a `containsMany` hosts field. Host is a `Person` card that `linksTo` the `Pet` card. So I couldn't add `Booking` card as a field on `Pet` card.

Found out that the `cardThunk` becomes `undefined` whenever there's a cyclic card relationship issue. These cycles are more complex than the ones we handled in the past because they occur through other cards.

Using the thunk format when writing fields gets around the error. However, in complex card relationship cases it's hard to determine which fields should be written in that format. One option is to have more complex queries in the card catalog, so that those cards that are far relatives don't show up on the catalog. The second option is converting each field that has a composite card to the thunk format, at least when added through the UI. Otherwise, throwing a better error than the type error thrown. Implemented the second option in this PR.

### CODE REVIEW

**Comment by habdelra** [comment]
yeah, maybe the thunk format should just be the default....

**Review by habdelra — APPROVED** [review]
:+1:

File: packages/runtime-common/module-syntax.ts [hunk 1]
```diff
@@ -236,7 +236,15 @@ function makeNewField(
...
   );
 
-  return `@${fieldDecorator.name} ${fieldName} = ${fieldTypeIdentifier.name}(${fieldCardIdentifier.name});`;
+  if (
+    fieldRef.module.startsWith(baseRealm.url) &&
+    fieldRef.name === 'default'
+  ) {
+    // primitive fields
+    return `@${fieldDecorator.name} ${fieldName} = ${fieldTypeIdentifier.name}(${fieldCardIdentifier.name});`;
+  }
```
Comment (habdelra) [inline packages/runtime-common/module-syntax.ts:245]: I just noticed this, this is not a very robust way to determine if a card is primitive. keep in mind that a primitive could live in the local realm, or have be a named export. it's just a coincidence that all primitives fit this particular pattern right now. A better idea might be to run the card and check for the primitive symbol....

Comment (burieberry) [reply packages/runtime-common/module-syntax.ts:245]: > I just noticed this, this is not a very robust way to determine if a card is primitive. keep in mind that a primitive could live in the local realm, or have be a named export. it's just a coincidence that all primitives fit this particular pattern right now. A better idea might be to run the card and check for the primitive symbol....

I tried loading the card and checking for primitive symbol but the primitive symbol is only available in card-api and loading a card is async.

Comment (habdelra) [reply packages/runtime-common/module-syntax.ts:245]: I'm hoping that my dead lock fix will fix this problem too...

**Pull request merged**